File: python/3 - Strings/1024_criptografia.py
def criptografia(texto):
    caracteres = []
    # Passada 1
    for caractere in texto:
        if caractere.isalpha():
            unidade = ord(caractere)
            unidadeCripto = chr(unidade + 3)
            caracteres.append(unidadeCripto)
        else:
            caracteres.append(caractere)
    # Passada 2
    caracteres = caracteres[::-1]
    # Passada 3
    primeiraMetade = caracteres[:len(caracteres)//2]
    segundaMetade = caracteres[len(caracteres)//2:]
    metadeCripto = []
    for caractere in segundaMetade:
        unidade = ord(caractere)
        unidadeCripto = chr(unidade - 1)
        metadeCripto.append(unidadeCripto)
    # Conclusão
    caracteres = primeiraMetade + metadeCripto
    mensagemCriptografada = "".join(caracteres)
    return mensagemCriptografada
    
testes = int(input())
for i in range(testes):
    mensagem = input()
    print(criptografia(mensagem))

File: java/javascript/2 - AD-HOC/1024_criptografia.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

function criptografia(texto) {
    let caracteres = [];

    // Passada 1
    for (let caractere of texto) {
        if (/[a-zA-Z]/.test(caractere)) {
            let unidade = caractere.charCodeAt(0);
            let unidade_cripto = String.fromCharCode(unidade + 3);
            caracteres.push(unidade_cripto);
        } else {
            caracteres.push(caractere);
        }
    }

    // Passada 2
    caracteres = caracteres.reverse();

    // Passada 3
    let meio = Math.floor(caracteres.length / 2);
    let primeira_metade = caracteres.slice(0, meio);
    let segunda_metade = caracteres.slice(meio);
    let metade_cripto = [];

    for (let caractere of segunda_metade) {
        let unidade = caractere.charCodeAt(0);
        let unidade_cripto = String.fromCharCode(unidade - 1);
        metade_cripto.push(unidade_cripto);
    }

    // Conclusão
    let caracteres_finais = primeira_metade.concat(metade_cripto);
    let mensagem_criptografada = caracteres_finais.join("");
    return mensagem_criptografada;
}

let testes = parseInt(lines[0]);
for (let i = 0; i < testes; i++) {
    let mensagem = lines[i + 1];
    console.log(criptografia(mensagem));
}